src/java_tools/buildjar/java/com/google/devtools/build/buildjar/VanillaJavaBuilder.java

```diff
@@ -103,15 +103,20 @@ private static int runPersistentWorker() {
         if (request == null) {
           break;
         }
+        WorkResponse response;
         try (VanillaJavaBuilder builder = new VanillaJavaBuilder()) {
           VanillaJavaBuilderResult result = builder.run(request.getArgumentsList());
-          WorkResponse response =
+          response =
```

Comment:
This seems a bit fragile / confusing to me. Wouldn't something like this also work and make it clearer that the try block is specifically for the VanillaJavaBuilder?

```java
VanillaJavaBuilderResult result;
try (...) {
  result = ...
}
WorkResponse response = ...;
```

Reply:
Sure, but I think you're commenting on the original code rather than the change.

Do you want to include this here or make it a follow-up?

Reply:
Let's make that a follow-up, get the actual bug fixed first.

Reply:
I'm not sure what you mean about the original code. If I click on "Files changed" in this PR, I'm seeing the final state of the change, and I commented on that. But... the GitHub UI always confuses me, so what am I missing?